File: krushi-ai/apps/web/src/lib/supabase.ts
import { createClient } from '@supabase/supabase-js';

// Provide explicit fallback values so the client doesn't crash during build or if .env is missing.
const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL || 'http://localhost:54321';
const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || 'ey_dummy_key_for_build';

export const supabase = createClient(supabaseUrl, supabaseAnonKey);

File: krushi-ai/apps/api/app/db/supabase.py
import os
from supabase import create_client, Client
from app.core.config import settings

# Provide explicit fallback to os.environ if pydantic-settings misses it during docker init
url = os.environ.get("SUPABASE_URL", settings.SUPABASE_URL)
key = os.environ.get("SUPABASE_KEY", settings.SUPABASE_KEY)

if not url or not key:
    print("WARNING: Supabase URL or Key is missing. Check your .env file or Docker environment variables.")

supabase: Client = create_client(url or "http://localhost:54321", key or "dummy")

def get_supabase_client() -> Client:
    return supabase
